pkg/controllers/cloud/aws/eks/reconcile.go

```
@@ -18,45 +18,247 @@ package eks
+			Name:    ComponentClusterBootstrap,
+			Message: "Accessing the eks cluster",
+			Status:  core.PendingStatus,
+		})
+
+		logger.Info("attempting to bootstrap the eks cluster")
+
+		boot, err := NewBootstrapClient(resource, client.Sess)
+		if err != nil {
+			logger.WithError(err).Error("trying to create bootstrap client")
+
+			return false, err
+		}
+		if err := boot.Run(ctx, t.mgr.GetClient()); err != nil {
+			logger.WithError(err).Error("trying to bootstrap eks cluster")
+
+			return false, err
+		}
+		resource.Status.Conditions.SetCondition(core.Component{
+			Name:    ComponentClusterBootstrap,
+			Message: "Successfully initialised the cluster",
+			Status:  core.SuccessStatus,
+		})
+
+		return false, nil
+	}()
 	if err != nil {
-		logger.WithError(err).Error("trying to update the resource status")
+		resource.Status.Status = core.FailureStatus
+	}
+
+	if err := t.mgr.GetClient().Status().Update(ctx, resource); err != nil {
+		logger.WithError(err).Error("updating the status of eks cluster")
 
 		return reconcile.Result{}, err
 	}
 
+	if err == nil {
+		if finalizer.NeedToAdd(resource) {
+			logger.Info("adding our finalizer to the team resource")
+
+			if err := finalizer.Add(resource); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			return reconcile.Result{Requeue: true}, nil
+		}
+	}
+
+	if requeue {
+		return reconcile.Result{Requeue: true}, nil
+	}
+
 	return reconcile.Result{}, nil
 }
+
+// GetCredentials returns the cloud credential
+func (t *eksCtrl) GetCredentials(ctx context.Context, cluster *eksv1alpha1.EKS, team string) (*eksv1alpha1.EKSCredentials, error) {
+	// @step: is the team permitted access to this credentials
+	permitted, err := t.Teams().Team(team).Allocations().IsPermitted(ctx, cluster.Spec.Credentials)
+	if err != nil {
+		log.WithError(err).Error("attempting to check for permission on credentials")
+
+		return nil, fmt.Errorf("attempting to check for permission on credentials")
```

we should return the error here

---

This is the error - the nil is the credentials object ;)

---

I meant the underlying error `err`, this will swallow it

---

I see, will fix